backend/src/modules/cart/service/cart.service.ts

```diff
@@ -49,12 +52,15 @@ export class CartService {
     }
     const oldItems = cart.items;
 
-    const update = {
-      userId: cart.userId,
-      items: this.mergeItems(oldItems, newItems),
-    };
+    await this.cartModel.updateOne(
+      { _id: cart.id },
+      {
+        userId: cart.userId,
+        items: this.mergeItems(oldItems, newItems),
+      },
+    );
 
-    return this.cartModel.findByIdAndUpdate(cart.id, update).exec();
+    return this.getUserCartItems(userId);
```

Comment:
same

Reply:
I use `this.getUserCartItems()` because i need to return cart items with price

Reply:
And you return cart items instead of whole cart
It can cause some error due to another response format